src/context/NotesContext.js

```diff
@@ -54,7 +69,68 @@ const NotesContextProvider = ({ children }) => {
       console.log(err);
     }
   };
-  const [counter, setCounter] = useState(0);
+  const editNote = async (noteItem) => {
+    console.log(noteItem);
+    setIsModalOpen(() => true);
+    setSelectedNote(() => noteItem);
+  };
+  const editTitle = (e) => {
+    console.log(e.target.value);
+    console.log({ ...selectedNote, title: e.target.value });
+    setSelectedNote(() => ({ ...selectedNote, title: e.target.value }));
+  };
+  const editBody = (e) => {
+    console.log(e.target.value);
+    setSelectedNote(() => ({ ...selectedNote, body: e.target.value }));
+  };
+  const updateNote = async (e) => {
+    e.preventDefault();
+    setCounter((counter) => counter + 1);
+    setIsPalleteOpen(() => false);
+    setIsModalOpen(() => false);
+    try {
+      const updatedNoteRes = await axios.post(
+        `/api/notes/${selectedNote._id}`,
+        {
+          note: selectedNote,
+        },
+        {
+          headers: {
+            authorization: encodedToken,
+          },
+        }
+      );
+      console.log(updatedNoteRes);
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  const changeCardColor = async (e) => {
+    console.log(e);
```

Comment:
if console output is not needed you can remove it as it is not necessary

Reply:
Yes I will remove it once I will complete P0 features.